source/credentials_provider_sts.c

```diff
@@ -0,0 +1,668 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+#include <aws/auth/credentials.h>
+#include <aws/auth/private/credentials_utils.h>
+#include <aws/auth/private/xml_parser.h>
+#include <aws/auth/signable.h>
+#include <aws/auth/signer.h>
+#include <aws/auth/signing_config.h>
+#include <aws/common/mutex.h>
+#include <aws/common/string.h>
+#include <aws/http/connection.h>
+#include <aws/http/connection_manager.h>
+#include <aws/http/request_response.h>
+#include <aws/io/socket.h>
+#include <aws/io/stream.h>
+#include <aws/io/tls_channel_handler.h>
+#include <aws/io/uri.h>
+
+#include <inttypes.h>
+
+static struct aws_byte_cursor s_host_name = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("host");
+static struct aws_byte_cursor s_host_name_val = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("sts.amazonaws.com");
+static struct aws_byte_cursor s_content_type = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("content-type");
+static struct aws_byte_cursor s_content_type_val =
+    AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("application/x-www-form-urlencoded");
+static struct aws_byte_cursor s_api_version = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("x-amz-api-version");
+static struct aws_byte_cursor s_api_version_val = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("2011-06-15");
+static struct aws_byte_cursor s_content_length = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("content-length");
+static struct aws_byte_cursor s_method = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("POST");
```

Comment:
You could also use this: https://github.com/awslabs/aws-c-http/blob/master/include/aws/http/http.h#L106